service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwin.java

```diff
@@ -216,7 +217,7 @@ public synchronized void updateTwin(DeviceTwinDevice device) throws IotHubExcept
         **Codes_SRS_DEVICETWIN_25_020: [** The function shall verify the response status and throw proper Exception **]**
          */
         Proxy proxy = options.getProxyOptions() != null ? options.getProxyOptions().getProxy() : null;
-        HttpResponse response = DeviceOperations.request(this.iotHubConnectionString, url, HttpMethod.PATCH, twinJson.getBytes(StandardCharsets.UTF_8), String.valueOf(requestId++),options.getHttpConnectTimeout(), options.getHttpReadTimeout(), proxy);
```

Comment:
The variable declaration was unused since this method has no return object

Reply:
no return object meaning void? how was this even compiling?

Reply:
Sorry, I meant that the HTTP response has no payload. It is just a 204. The method still returns an HTTP response object, we just don't do anything with it